JoyboyCommunity/src/shared/components/Post.tsx

```diff
@@ -62,84 +168,111 @@ export const Post: React.FC<PostProps> = (props) => {
   const handleNavigateToPostDetails = () => {
     navigation.navigate('PostDetail', {postId: event?.id, post: event});
   };
+
   /** @TODO react in Nostr */
   const handleReact = () => {};
 
+  /** @TODO comment in Nostr */
+  const toggleLike = () => {
+    setIsLiked((prev) => !prev);
+    setLikes((prev) => {
+      const newLikes = isLiked ? prev - 1 : prev + 1;
+      textOpacity.value = newLikes > 0 ? 1 : 0;
+      iconPosition.value = newLikes > 0 ? 0 : 6;
```

Comment:
maybe we can use a single animation value and use interpolation for different values. For example:
```tsx
const animation = useSharedValue(0);
const animatedStyle = useAnimatedStyle(() => ({
  opacity: animation.value,
  transform: [{translateX: interpolate(animation.value, [0, 1], [0, 6])}]
}));
```